src/mapengine/leaflet/js/leaflet-map-engine.debug.js

```diff
@@ -1346,74 +1399,77 @@ emp.engineDefs.leafletMapEngine = function (args) {
-                        }
-                    }
-                }
+      var item = transaction.items[0],
+          oMap = instanceInterface.getLeafletMap(),
+          level = oMap.getZoom(),
+          bZoom = ((item.zoom === true) ? true : ((typeof (item.zoom) === 'number') && (item.zoom > 0)) ? true : false),
+          southWest,
+          northEast,
+          bounds,
+          oEmpObject,
+          oEmpBounds,
+          oCenter;
+
+      try {
+        if (instanceInterface.getLockState() === emp3.api.enums.MapMotionLockEnum.NO_MOTION) {
+          throw new Error("The current lock state is set to 'NO_MOTION'.");
+        } else {
+          if (item.location) {
+            oMap.setView(new L.LatLng(item.location.lat, item.location.lon), level, {animate: false});
+            if (item.range && !isNaN(item.range)) {
+              level = instanceInterface.getLevel(item.range * instanceInterface.rangeToScaleMultiplier);
+              oMap.setZoom(level, {animate: false});
             }
-//            else if (item.range || (item.zoom !== false)) {
-            else if (item.range) {
-                level = instanceInterface.getLevel(item.range * instanceInterface.rangeToScaleMultiplier);
-                oMap.setZoom(level, {animate: false});
+          } else if (item.bounds) {
+            southWest = L.latLng(item.bounds.south, item.bounds.west);
+            northEast = L.latLng(item.bounds.north, item.bounds.east);
+            bounds = L.latLngBounds(southWest, northEast);
+            oMap.fitBounds(bounds);
+            if (item.range && !isNaN(item.range)) {
+              level = instanceInterface.getLevel(item.range * instanceInterface.rangeToScaleMultiplier);
+              oMap.setZoom(level, {animate: false});
             }
-
-            instanceInterface.getView(transaction.items[0]);
-        } catch (err) {
-            transaction.fail([new emp.typeLibrary.Error({
+          } else if (item.featureId || item.overlayId) {
+            oEmpObject = instanceInterface.mapEngObjectList[item.coreId];
+            if (oEmpObject !== undefined) {
+              if (oEmpObject.getObjectType() === leafLet.typeLibrary.objectType.OVERLAY) {
+                // We need to check to see if it has features in it.
+                if (!oEmpObject.hasFeatures()) {
+                  transaction.fail(new emp.typeLibrary.Error({
                     coreId: item.coreId,
-                    message: "An error occurred while attempting to set the map view: ",
                     level: emp.typeLibrary.Error.level.MINOR,
-                    jsError: err
-                })]);
+                    message: "Overlay " + oEmpObject.options.name + " is empty."
+                  }));
+                  return;
+                }
+              }
+              oEmpBounds = oEmpObject.getFeatureBounds();
+              if (oEmpBounds) {
+                if (bZoom) {
+                  if (oEmpBounds.getCenterWidth() < 16093) {
```

Comment:
Can you move the value into a named variable? Could be unclear what 16093 refers to